packages/venia-ui/lib/targets/makeRoutesTarget.js

```diff
@@ -11,10 +11,14 @@ function makeRoutesTarget(venia) {
 function addRoutes(routeList, routes) {
     for (const route of routes) {
         const AddedRoute = routeList.addReactLazyImport(route.path, route.name);
+
+        const routePath = Array.isArray(route.pattern) ? 
+        `{[${route.pattern.map(path => `"${path}"`).join(", ")}]}` : `"${route.pattern}"`;
+
         routeList.prependJSX(
             'Switch',
             `<Route ${route.exact ? 'exact ' : ''}path="${
-                route.pattern
+                routePath
             }"><${AddedRoute}/></Route>`
```

Comment:
The `routePath` creation now adds the double-quotes or brackets when necessary so you should remove them from line 20/22.